### PR TITLE
Simplify state update callback 

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- Deprecate updateListener BS
- code clean up

### _How did you test your changes?_

existing unit test

### CODE REVIEW

File: packages/wallet-sdk/src/core/provider/interface.ts [hunk 1]
```diff
@@ -13,23 +13,25 @@ export interface ProviderRpcError extends Error {
   data?: unknown;
 }
 
-interface ProviderMessage {
-  type: string;
-  data: unknown;
-}
-
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/provider/interface.ts:20]: not being used

**Review by Vt2049 — COMMENTED** [review]
..

**Comment by Vt2049** [comment]
> ### _สรุป_
> ### _คุณทดสอบการเปลี่ยนแปลงของคุณอย่างไร?_

k

**Comment by Vt2049** [comment]
k

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 2]
```diff
@@ -63,7 +62,7 @@ describe('Event handling', () => {
     provider.on('chainChanged', chainChangedListener);
 
     await provider.request({ method: 'eth_requestAccounts' });
-    updateListener.onChainIdUpdate(1);
+    callback('chainChanged', '0x1');
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:65]: it needs to be hex string
image

File: packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts [hunk 3]
```diff
@@ -28,12 +33,14 @@ export class ExtensionSigner implements Signer {
...
-    this.extensionProvider.on('accountsChanged', (accounts) =>
-      this.updateListener.onAccountsUpdate(accounts as AddressString[])
+    const events: Parameters<ProviderEventCallback>[0][] = [
+      'connect',
+      'chainChanged',
+      'accountsChanged',
+      'disconnect',
+    ];
+    events.forEach((event) =>
+      this.extensionProvider.on(event, (data) => this.callback(event, data))
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts:43]: 🤮

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 4]
```diff
@@ -124,10 +122,7 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
       signerType,
       metadata: this.metadata,
       communicator: this.communicator,
-      updateListener: {
-        onAccountsUpdate: (accounts) => this.emit('accountsChanged', accounts),
-        onChainIdUpdate: (id) => this.emit('chainChanged', hexStringFromNumber(id)),
-      },
+      callback: this.emit.bind(this),
```
Comment (nateReiners) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:126]: <3

File: packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts (hunk 3 above)
Comment (nateReiners) [reply packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts:43]: disconnect might be fired twice too 🤔  https://github.com/coinbase/coinbase-wallet-sdk/pull/1350/files#diff-bb3b1961b89bdeabb2d5abe40c8646da1dda20ca67ded4d932c32002e76d70daL105

**Review by Vt2049 — COMMENTED** [review]
ok

**Review by Vt2049 — COMMENTED** [review]
,.

**Review by Vt2049 — COMMENTED** [review]
..

**Review by nateReiners — APPROVED** [review]
left comment about the events but looks great otherwise

File: packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts (hunk 3 above)
Comment (bangtoven) [reply packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts:43]: @nateReiners you are right.
but i think we are about to remove this ExtensionSigner soon.

**Pull request merged**